app/src/main/java/de/westnordost/streetcomplete/data/osm/osmquests/OsmQuest.kt

```diff
@@ -27,16 +27,23 @@ data class OsmQuest(
     override val type: QuestType<*> get() = osmElementQuestType
 
     override val markerLocations: Collection<LatLon> get() {
-        if (osmElementQuestType.hasMarkersAtEnds && geometry is ElementPolylinesGeometry) {
+        if (geometry is ElementPolylinesGeometry) {
             val polyline = geometry.polylines[0]
             val length = polyline.measuredLength()
-            if (length > 15 * 4) {
-                return listOf(
-                    polyline.pointOnPolylineFromStart(15.0)!!,
-                    polyline.pointOnPolylineFromEnd(15.0)!!
-                )
+            if ((osmElementQuestType.hasMarkersAtEnds && length > MARKER_FROM_END_DISTANCE * 4)
+                || (length > MAXIMUM_MARKER_DISTANCE + 2 * MARKER_FROM_END_DISTANCE)) {
```

Comment:
```suggestion
            val minDistanceForAddingMultipleMarkers =
                if (osmElementQuestType.hasMarkersAtEnds) MARKER_FROM_END_DISTANCE * 4
                else MAXIMUM_MARKER_DISTANCE + 2 * MARKER_FROM_END_DISTANCE
            if (length > minDistanceForAddingMultipleMarkers) {
```
I think this might also help readability here (didn't test if this compiles/runs)

Reply:
If you take my other suggestion, you could pull the `start`, `end`, and `middleLength` calculations outside the `if` statement and simplify this a little more

```suggestion
            val minDistanceForAddingMultipleMarkers =
                if (osmElementQuestType.hasMarkersAtEnds) MARKER_FROM_END_DISTANCE * 2 else MAXIMUM_MARKER_DISTANCE
            if (middleLength > minDistanceForAddingMultipleMarkers) {
```

Reply:
I have done something similar to this now.

Reply:
Looks good to me.